packages/amplify-cli/src/attach-backend.ts

```diff
@@ -105,22 +112,19 @@ const backupAmplifyFolder = (): void => {
     const backupAmplifyDirPath = path.join(projectPath, backupAmplifyDirName);
 
     if (fs.existsSync(backupAmplifyDirPath)) {
-      const error = new Error(`Backup folder at ${backupAmplifyDirPath} already exists, remove the folder and retry the operation.`);
-
-      error.name = 'BackupFolderAlreadyExist';
-      error.stack = undefined;
-
-      throw error;
+      throw amplifyErrorWithTroubleshootingLink('DirectoryAlreadyExistsError', {
+        message: `Backup folder at ${backupAmplifyDirPath} already exists, remove the folder and retry the operation.`,
+      });
     }
     try {
       fs.moveSync(amplifyDirPath, backupAmplifyDirPath);
     } catch (e) {
-      if (e.code === 'EPERM') {
-        throw new Error(
-          'Could not attach the backend to the project. Ensure that there are no applications locking the `amplify` folder and try again',
-        );
-      }
-      throw e;
+      throw amplifyErrorWithTroubleshootingLink('DirectoryError', {
+        message: `Could not attach the backend to the project.`,
+        resolution: 'Ensure that there are no applications locking the `amplify` folder and try again.',
+        details: e.message,
+        stack: e.stack,
+      });
     }
```

Comment:
if we only know a resolution for "EPERM", we should keep the IF statement here in the resolution message;
resolution: e.code == 'EPERM' ? "Ensure that..." : "An error occurred attaching the backend.."

Reply:
'An error occurred attaching the backend' is not a valid resolution, what I can do is to keep the if and throw a fault for anything other that `EPERM`

Reply:
Ok; that would work. We provide resolution IF we have it; otherwise, throw the fault